resources/timerbox.ts

```diff
@@ -408,14 +423,27 @@ export default class TimerBox extends HTMLElement {
     clearTimeout(this._timer ?? 0);
     this._timer = null;
     this.classList.remove('expired');
+    this._notifyThresholdCallbacks = true;
 
     this._start = new Date().getTime();
+
+    for (const f of this._onResetCallbacks)
+      setTimeout(f, 0);
```

Comment:
Can I ask what `setTimeout` is doing here vs just calling these callbacks directly? Is there some ordering constraint here or are you just making sure they run outside of the advance/reset callstacks?

Reply:
Pretty much. It is probably irrelevant, but as a callback could in theory do anything, including changing properties of the current timerbox, I wanted to ensure the current operation finishes without the need to potentially think about side effects from any of the callbacks.